frontend/views/containers/proposals/ProposalsWidget.vue

```diff
@@ -134,14 +136,26 @@ export default ({
     }
   },
   methods: {
-    onProposalArchived (hashWithProp) {
-      this.ephemeral.archivedProposals.unshift(hashWithProp)
+    async updateArchivedProposals (hashWithProp) {
+      const key = `proposals/${this.ourUsername}/${this.currentGroupId}`
+      const archivedProposals = await sbp('gi.db/archive/load', key) || []
+      this.ephemeral.archivedProposals = archivedProposals
+        .filter(([hash, obj]) => Date.now() - new Date(obj.dateClosed).getTime() < DAYS_MILLIS)
+        .sort(([hash1, obj1], [hash2, obj2]) => new Date(obj2.dateClosed).getTime() - new Date(obj1.dateClosed).getTime())
+
       // after a day, remove it from the list
-      this.ephemeral.timeouts.push(setTimeout(() => {
-        this.ephemeral.archivedProposals = this.ephemeral.archivedProposals.filter(x => {
-          return x[0] !== hashWithProp[0]
-        })
-      }, 1000 * 60 * 60 * 24)) // 1 day
+      this.clearTimeouts()
+      for (const [hash, obj] of this.ephemeral.archivedProposals) {
+        const dateClosed = new Date(obj.dateClosed).getTime()
+        this.ephemeral.timeouts.push(setTimeout(() => {
+          this.ephemeral.archivedProposals = this.ephemeral.archivedProposals.filter(x => x[0] !== hash)
+        }, dateClosed - Date.now() + DAYS_MILLIS))
+      }
+    },
```

Comment:
Here `hashWithProp` is no longer being used... is that correct? If so, should it be removed from the function parameters?

Is this line correct?

```js
this.ephemeral.archivedProposals = this.ephemeral.archivedProposals.filter(x => x[0] !== hash)
```

It used to check against `hashWithProp[0]`, but now it doesn't.

What is this condition doing instead? It looks like it might be removing all of the proposals...?

Reply:
> Here `hashWithProp` is no longer being used... is that correct? If so, should it be removed from the function parameters?

I didn't want to use that parameter, but I started thinking we should use that parameter to ignore the redundant operations. But one more parameter is needed. `groupId`. Nice catch, @taoeffect

> Is this line correct?

Yes

> What is this condition doing instead? It looks like it might be removing all of the proposals...?

Proposals which are archived in the last 24 hours should be removed automatically when 24 hours are passed from when they are archived. The timers are doing this.